main.cpp

```diff
@@ -41,23 +41,310 @@ If you need inspiration for what to write, take a look at previously approved st
+        ~Sandball();
+        void fillBall(int howLong);
+        void changeFilling();
+        void cutTail(int cutAmount);
+    };
+
+    Ball();
+    ~Ball();
+    void getHeavier(bool soakInWater, int soakingTime = 3);
+    double getVolume(bool isHollow);
+    void addDimples(int howManyToAdd);
+};
+
+Ball::Ball() : 
+ballRadius(2.5),
+ballWeight(4.2f){}
+
+Ball::~Ball()
+{
+    std::cout << "Ball Destroyed" << std::endl;
+}
+
+void Ball::getHeavier(bool soakInWater, int soakingTime)
+{
+    if(soakInWater)
+    {
+        while(soakingTime > 0)
+        {
+            //std::cout << "Soaked Ball" << std::endl;
+            --soakingTime;
+            ballWeight += 0.2f;
+        }
+    }
+}
+
+double Ball::getVolume(bool isHollow)
+{
+    if(isHollow)
+    {
+        return 0.0;
+    }
+    double currentVolume;
+    currentVolume = (4/3)*(3.14159)*(ballRadius * ballRadius);
+    return currentVolume;
+}
+
+void Ball::addDimples(int howManyToAdd)
+{
+    numberOfDimples += howManyToAdd;
+}
+
+Ball::Sandball::Sandball() :
+    sandWeight(5.1f),
+    fineSandFilling(true),
+    lengthOfTail(15){}
+
+Ball::Sandball::~Sandball()
+{
+    std::cout << "Sandball Destroyed" << std::endl;
+}
+
+void Ball::Sandball::fillBall(int howLong)
+{
+    while(howLong > 0)
+    {
+        percentFilled++;
+        sandWeight = sandWeight + 0.1f;        
+        --howLong;
+    }
+}
+
+void Ball::Sandball::changeFilling()
+{
+    (fineSandFilling) ? (fineSandFilling = false) : (fineSandFilling = true);
+}
+
+void Ball::Sandball::cutTail(int cutAmount)
+{
+    lengthOfTail -= cutAmount;
+}
 
 /*
  UDT 3:
  */
 
+struct Bird
+{
+    float birdWeight;
+    double groundSpeed{3.4};
+    double distanceWaddled{104.4};
+    bool webbedFeet = false;
+    bool sharpBill;
+
+    Bird();
+    ~Bird();
+    bool canSwim(bool birdFloats);
+    void eatMeal(float mealWeight, bool mealInShell);
+    void waddle(int timeWaddling);
+
+    struct FlyingBird
+    {
+        float flightSpeed;
+        int flapsPerMinute;
+        double migrationDistance;
+        bool flysInFlocks = true;
+        float lengthOfWings{4.5f};
+
+        FlyingBird();
+        ~FlyingBird();
+        void migrate(bool flySouth, double tripDistance);
+        int flapsPerTrip(int minutesPerTrip);
+        void growWings(float growthAmount);
+
+    };
+};
+
+Bird::Bird() : 
+birdWeight(10.f),
+sharpBill(false)
+{}
+
+Bird::~Bird()
+{
+    std::cout << "Bird Destroyed" << std::endl;
+}
+
+bool Bird::canSwim(bool birdFloats)
+{
+    if(birdFloats)
+    {
+        if(webbedFeet)
+        {
+            return true;
+        }
+        return false;
+    }
+    return false;
+}
+
+void Bird::eatMeal(float mealWeight, bool mealInShell)
+{
+    if(mealInShell)
+    {
+        if(sharpBill)
+        {
+            birdWeight += mealWeight;
+        }
+    }
+}
+
+void Bird::waddle(int timeWaddling)
+{
+    while(timeWaddling > 0)
+    {
+        distanceWaddled += groundSpeed;
+    }
```

Comment:
you never change `timeWaddling`, which means this will loop forever if you call the function with something other than 0.
- [ ] revise.